sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/FunctionAppImpl.java

```diff
@@ -698,6 +700,31 @@ public FunctionAppImpl withMinReplicas(int minReplicas) {
         return this;
     }
 
+    @Override
+    public Mono<Map<String, AppSetting>> getAppSettingsAsync() {
+        if (isFunctionAppOnACA()) {
+            // current function app on ACA doesn't support deployment slot, so appSettings sticky is false
+            return listAppSettings()
+                .map(
+                    appSettingsInner ->
+                        appSettingsInner
+                            .properties()
+                            .entrySet()
+                            .stream()
+                            .collect(
+                                Collectors
+                                    .toMap(
+                                        Map.Entry::getKey,
+                                        entry ->
+                                            new AppSettingImpl(
+                                                entry.getKey(),
+                                                entry.getValue(),
+                                                false))));
+        } else {
+            return super.getAppSettingsAsync();
+        }
```

Comment:
diff from parent is that parent will also call listSlotConfiguration to set appSetting's stickness accordingly.

Function App on ACA doesn't support deployment slot, listSlotConfiguration will throw exception. So always set stickness to false.